File: _BHCVRP_Python_Version/BHCVRP_Python_Version/cujae_inf_citi_om/exceptions/WithoutCapacityException.py
class WithoutCapacityException(Exception):
    def __init__(self, capacity=None):
        if capacity is not None:
            self.capacity = capacity
        
    def __init__(self, message=None):
        if message is not None:
            super().__init__(message)

    def get_capacity(self):
        return getattr(self, 'capacity', None)

File: BHCVRP_Java_Version/src/cujae/inf/citi/om/exceptions/WithoutCapacityException.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cujae.inf.citi.om.exceptions;

/**
 *
 * @author nanda
 */

//Clase que maneja la excepción lanzada cuando no existe capacidad.

public class WithoutCapacityException extends Exception {
    private int capacity;
    
    public WithoutCapacityException(int capacity){
        this.capacity = capacity;
    
    }
    
    public int getCapacity(){
        return capacity;
    }
    
    public WithoutCapacityException(String message){   
        super(message);
    }
}

